cmd/kubeadm/app/cmd/certs.go

```diff
@@ -134,7 +134,7 @@ func (o *genCSRConfig) addFlagSet(flagSet *pflag.FlagSet) {
 func (o *genCSRConfig) load() (err error) {
 	o.kubeadmConfig, err = configutil.LoadOrDefaultInitConfiguration(
 		o.kubeadmConfigPath,
-		cmdutil.DefaultInitConfiguration(),
+		&kubeadmapiv1.InitConfiguration{},
```

Comment:
are we sure this (and similar changes) will not cause the autodetection to start happening for commands that don't need it? 
we might be lacking full unit test coverage around the cri socket detection problem.

Reply:
> we might be lacking full unit test coverage around the cri socket detection problem.

@neolit123 To be honest, yes. 
Currently, we have to check it manually. That's what I've done.

For this change, as we have passed the `skipCRIDetect = true` parameter to `configutil.LoadOrDefaultInitConfiguration()`, the auto-detection will not happen. 
And the other changes are similar.
https://github.com/kubernetes/kubernetes/blob/c3bf541edeefc1cfcf24ef5e4cb5481e529c78f8/cmd/kubeadm/app/cmd/certs.go#L135-L140

Reply:
ok, LGTM.
 i leave this to your judgement.